src/simsopt/geo/curvexyzfourier.py

```diff
@@ -72,16 +74,77 @@ def set_dofs(self, dofs):
         sopp.CurveXYZFourier.set_dofs(self, dofs)
 
     @staticmethod
-    def load_curves_from_file(filename, order=None, ppp=20, delimiter=','):
+    def load_curves_from_file(filename: str, order: int, ppp=20, delimiter=',', maxiter=2000, tol=1.48e-8):
         """
-        This function loads a file containing Fourier coefficients for several coils.
-        The file is expected to have :mod:`6*num_coils` many columns, and :mod:`order+1` many rows.
+        This function loads a file containing Fourier coefficients
+        or the cartesian coordinates for several coils.
+        For the Fourier coefficient case, the file is expected to have :mod:`6*num_coils` many columns, and :mod:`order+1` many rows.
         The columns are in the following order,
 
             sin_x_coil1, cos_x_coil1, sin_y_coil1, cos_y_coil1, sin_z_coil1, cos_z_coil1, sin_x_coil2, cos_x_coil2, sin_y_coil2, cos_y_coil2, sin_z_coil2, cos_z_coil2,  ...
 
+        For the cartesian case, The format is introduced at
+        https://princetonuniversity.github.io/STELLOPT/MAKEGRID
+
+        Args:
+            filename: file to load.
+            order: maximum order in the fourier expansion.
+            ppp: point-per-period: number of quadrature points per period.
+            delimiter: in the case of a file containing Fourier coefficients the delimiter between coefficients.
+            maxiter: maximum number of iterations for the integration method.
+            tol: minimum tolerance for the convergence of the integration method. 
+        Returns:
+            A list of objects CurveXYZFourier with the coefficients given by the file.
+
         """
-        coil_data = np.loadtxt(filename, delimiter=delimiter)
+        if "coils." in str(filename):
```

Comment:
`"coils."` is very arbitrary. Someone could change the filename. Do you mean `".coils"`?